react/components/ButtonPlain/index.tsx

```diff
@@ -0,0 +1,224 @@
+  onFocus: PropTypes.func,
+  /** onBlur event */
+  onBlur: PropTypes.func,
+  /** Link spec */
+  target: PropTypes.string,
+  /** Link spec */
+  rel: PropTypes.string,
+  /** Link spec */
+  referrerPolicy: PropTypes.string,
+  /** Link spec */
+  download: PropTypes.string,
+  /** Disables label wrapping */
+  noWrap: PropTypes.bool,
+}
+
+type Props = InferProps<typeof propTypes>
+
+const ButtonPlain: FunctionComponent<Props> = props => {
+  const handleClick = (event: React.MouseEvent) => {
+    !props.disabled && !props.isLoading && props.onClick && props.onClick(event)
+  }
+
+  useEffect(() => {
+    if (props.icon) {
+      console.warn(
+        'Button: The prop "icon" of the "Button" component has been deprecated, and will be removed in a future version. Please use the component "ButtonWithIcon" instead'
+      )
+    }
+  }, [])
+
+  const {
+    size,
+    variation,
+    children,
+    icon,
+    isLoading,
+    href,
+    target,
+    rel,
+    referrerPolicy,
+    download,
+    noWrap,
+  } = props
+
+  const disabled = props.disabled || isLoading
+  const iconOnly = icon || props.iconOnly
+  const horizontalPadding = iconOnly ? 1 : 2
+  const horizontalCompensation = `nr${horizontalPadding} nl${horizontalPadding} `
+
+  const classes = classNames(
+    'vtex-button bw1 ba fw5 v-mid relative pa0 br2 bn',
+    horizontalCompensation,
+    {
+      't-action--small': size === SIZE.SMALL,
+      't-action': size === SIZE.REGULAR,
+      't-action--large': size === SIZE.LARGE,
+    },
+    { 'icon-button dib': iconOnly },
+    {
+      'bg-transparent b--transparent c-disabled': disabled,
+      'bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary':
+        !disabled && variation === VARIATION.PLAIN,
+      'bg-transparent b--transparent c-on-base--inverted':
+        !disabled && variation === VARIATION.INVERTED,
+      'bg-transparent b--transparent c-danger hover-bg-danger--faded hover-b--danger--faded':
+        !disabled && variation === VARIATION.DANGER,
+    },
+    { pointer: !disabled },
+    { 'inline-flex items-center no-underline': href }
+  )
+
+  const labelClasses = classNames(
+    'flex items-center justify-center h-100 pv1 ttn ',
+    `ph${horizontalPadding}`,
+    { nowrap: noWrap }
+  )
+
+  const style: { fontSize?: number } = {}
+
+  if (iconOnly) {
+    style.fontSize = 0
+  }
+
+  const linkModeProps = {
+    target,
+    rel,
+    referrerPolicy,
+    download,
+  }
+
+  const Element = href ? 'a' : 'button'
+
+  return (
+    <Element
+      id={props.id}
+      data-testid={props.testId}
+      autoFocus={iconOnly ? undefined : props.autoFocus}
+      disabled={iconOnly ? undefined : props.disabled}
+      name={iconOnly ? undefined : props.name}
+      value={iconOnly ? undefined : props.value}
+      tabIndex={0}
+      className={classes}
+      href={href}
+      onClick={handleClick}
+      onMouseEnter={props.onMouseEnter}
```

Comment:
I think it would be nice if you get `const {  ...rest } = props` and pass it here, it would make it easier to pass some prop that we missed to add and you wouldn't have to pass all these eventlisteners

Reply:
Yes we will do that in the next iteration of the Styleguide, ONDA. But for now we want to limit the abilities of each component deliberately.